assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/vision/track_predict.py

```diff
@@ -0,0 +1,196 @@
+
+            try:
+                model_config_path = context.artifacts[MMDetLiterals.CONFIG_PATH]
+                model_weights_path = context.artifacts[MMDetLiterals.WEIGHTS_PATH]
+
+                if not torch.cuda.is_available():
+                    raise RuntimeError("CUDA is not available. GPU needed for MOT inference!")
+                self._config = mmcv.Config.fromfile(model_config_path)
+                self._model = init_model(self._config, model_weights_path)
+
+                print("Model loaded successfully")
+            except Exception:
+                print("Failed to load the the model.")
+                raise
+
+        else:
+            raise ValueError(f"invalid task type {self._task_type}."
+                             f"Supported tasks: video-multi-object-tracking")
+
+    def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.DataFrame) -> pd.DataFrame:
+        """Perform inference on the input data.
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        :param input_data: Input images for prediction
+        :type input_data: Pandas DataFrame with a first column name ["image"] of images where each
+        image is in base64 String format.
+        :return: Output of inferencing
+        :rtype: Pandas DataFrame with columns ["boxes"] for object detection
+        """
+        # process the images in image column
+        processed_videos = input_data.loc[:, [MLflowSchemaLiterals.INPUT_COLUMN_VIDEO]].apply(
+            axis=1, func=_process_video
+        )
+        all_predictions = []
+        for _, video_url in processed_videos.iteritems():
+            video_frames = self.video_reader(video_url)
+            for j, img in enumerate(video_frames):
+                result = self._inference_detector(self._model, img, frame_id=j)
+                result = self._parse_mot_output(result, j, video_url)
+                all_predictions.append(result)
+
+        return pd.DataFrame(all_predictions)
+
+    def _parse_mot_output(self, result, frame_id, video_url) -> List[Dict]:
+        """Parse the output of inference_mot() to a list of dictionaries.
+        param result: output of inference_mot()
+        type result: tuple
+        param frame_id: frame id of the current frame
+        type frame_id: int
+        param video_url: url of the video
+        type video_url: str
+        return: dictionary of current image predictions
+        """
+        print(result.keys())
+        det_bboxes = result[MOTLiterals.DET_BBOXES]
+        track_bboxes = result[MOTLiterals.TRACK_BBOXES]
+
+        curimage_preds = {MOTLiterals.DET_BBOXES: [], MOTLiterals.TRACK_BBOXES: [],
+                          MOTLiterals.FRAME_ID: frame_id, MOTLiterals.VIDEO_URL: video_url}
+        for label, bboxes in enumerate(det_bboxes):
+            for bbox in bboxes:
```

Comment:
In the corresponding function in mmtrack_utils.py, we have a check for if label >= 0. Is that not required here ?
Also, the inner for-loop "for bbox in bboxes:" is missing in mmtrack_utils.py. Is there any difference in det_bboxes format between the two

Reply:
yes, the output from the two are different. for this one, there are only two: "track_bboxes", "det_bboxes", but in the mlflow wrapper there are four: "track_bboxes", "track_labels", "det_bboxes", "det_labels". In short, it's organized version. So we will have to test if the label is >=0. here we can just look through the ndarray, and parse the output.

Reply:
Thanks for the clarification :)